model-optimizer/extensions/middle/StridedSliceNormalizer.py

```diff
@@ -0,0 +1,251 @@
+            # Until now it was not possible to set correct permutations
+            PermuteInputs().set_input_permutation(node.in_node(1), node, 'input:1', 'slice', 'dim_size')
+            PermuteInputs().set_input_permutation(node.in_node(2), node, 'input:2', 'slice', 'dim_size')
+            PermuteInputs().set_input_permutation(node.in_node(3), node, 'input:3', 'slice', 'dim_size')
+
+    @staticmethod
+    def normalize_strided_slice(graph: Graph, node: Node):
+        input_shape = node.in_port(0).data.get_shape()
+        input_rank = len(input_shape)
+        begin, _, _ = StridedSlice.validate_inputs_and_get_args(node)
+        slice_rank = len(begin)
+
+        StridedSlice.align_mask_with_slice_rank(node, slice_rank)  # if StridedSlice is created after partial_infer
+        StridedSliceNormalizer.normalize_slices_attr(node)
+
+        num_insertions = input_rank - slice_rank + np.count_nonzero(node.new_axis_mask)
+        assert num_insertions >= 0, 'slice_rank - num_new_axis must <= input rank. Got instead: ' \
+                                    'input_rank = {}, slice_rank = {}, num_new_axis = {}'. \
+            format(input_rank, slice_rank, np.count_nonzero(node.new_axis_mask))
+
+        if np.any(node.ellipsis_mask):
+            assert np.count_nonzero(node.ellipsis_mask) == 1, 'only one ellipsis_mask nonzero value is allowed'
+            ellipsis_start = np.nonzero(node.ellipsis_mask)[0][0]
+            # since we don't expect values in begin and end: take the whole range along ellipsis_start
+            node.begin_mask[ellipsis_start] = 0
+            node.end_mask[ellipsis_start] = 0
+            node.ellipsis_mask[ellipsis_start] = 0
+            insertion_start_idx = ellipsis_start + 1
+
+            StridedSliceNormalizer.unroll_ellipsis_for_inputs(graph, node, ellipsis_start, num_insertions)
+        elif num_insertions > 0:
+            insertion_start_idx = slice_rank  # insert blank values to mask ends
+            StridedSliceNormalizer.extend_inputs(node, num_insertions)
+
+        if num_insertions > 0:
+            # insert blank values for ellipsis unrolling and extending
+            for mask_name in StridedSlice.get_mask_names():
+                node[mask_name] = np.insert(node[mask_name], insertion_start_idx, [0] * num_insertions).astype(int)
+
+    @staticmethod
+    def unroll_ellipsis_for_inputs(graph: Graph, node: Node, ellipsis_start: int, num_insertions: int):
+        node_name = node.soft_get('name', node.id)
+
+        for i, input_name in [(1, 'begin'), (2, 'end'), (3, 'strides')]:
+            blank_values_arr = np.zeros(num_insertions) if input_name != 'strides' else np.ones(num_insertions)
+            blank_values_node = Const(graph, {'name': node_name + '/const_to_unroll_{}_ellipsis'.format(input_name),
+                                              'value': int64_array(blank_values_arr)}).create_node()
+
+            if input_name == 'strides' and node.in_port(3).disconnected():
```

Comment:
`if i == 3 and node.in_port(3).disconnected()` ?

Reply:
Done